kernel/kernel-api/src/main/java/io/delta/kernel/engine/CommitCoordinatorClientHandler.java

```diff
@@ -0,0 +1,152 @@
+     *
+     * The returned latestTableVersion is the maximum commit version ratified by the commit
+     * coordinator. Note that returning latestTableVersion as -1 is acceptable only if the commit
+     * coordinator never ratified any version, i.e. it never accepted any unbackfilled commit.
+     *
+     * @param tablePath The path to the delta log of the table for which the unbackfilled
+     *                  commits should be retrieved.
+     * @param tableConf The table configuration that was returned by the commit coordinator
+     *                  during registration.
+     * @param startVersion The minimum version of the commit that should be returned. Can be null.
+     * @param endVersion The maximum version of the commit that should be returned. Can be null.
+     * @return GetCommitsResponse which has a list of {@link Commit}s and the latestTableVersion
+     *         which is tracked by {@link CommitCoordinatorClientHandler}.
+     */
+    GetCommitsResponse getCommits(
+            String tablePath,
+            Map<String, String> tableConf,
+            Long startVersion,
+            Long endVersion);
+
+    /**
+     * API to ask the commit coordinator client to backfill all commits up to {@code version}
+     * and notify the commit coordinator.
+     *
+     * If this API returns successfully, that means the backfill must have been completed, although
+     * the commit coordinator may not be aware of it yet.
+     *
+     * @param logPath The path to the delta log of the table that should be backfilled.
+     * @param tableConf The table configuration that was returned by the commit coordinator
+     *                  during registration.
+     * @param version The version till which the commit coordinator client should backfill.
+     * @param lastKnownBackfilledVersion The last known version that was backfilled before this API
+     *                                   was called. If it is None or invalid, then the commit
+     *                                   coordinator client should backfill from the beginning of
+     *                                   the table. Can be null.
+     * @throws IOException if the backfill failed.
+     */
+    void backfillToVersion(
+            String logPath,
+            Map<String, String> tableConf,
+            long version,
+            Long lastKnownBackfilledVersion) throws IOException;
+
+    /**
+     * Determines whether this CommitCoordinatorClientHandler is semantically equal to another
+     * CommitCoordinatorClientHandler.
+     * <p>
+     * Semantic equality is determined by each CommitCoordinatorClientHandler implementation based
+     * on whether the two instances can be used interchangeably when invoking any of the
+     * CommitCoordinatorClientHandler APIs, such as {@link #commit}, {@link #getCommits}, etc. For
+     * example, both instances might be pointing to the same underlying endpoint.
+     */
```

Comment:
```suggestion
/**
 * Compares this {@link CommitCoordinatorClientHandler} for semantic equality with another instance.
 *
 * <p>Semantic equality is defined by the specific implementation of {@link CommitCoordinatorClientHandler}. 
 * Two instances are considered semantically equal if they can be used interchangeably for any 
 * {@link CommitCoordinatorClientHandler} API, including but not limited to:</p>
 *
 * <ul>
 *   <li>{@link #commit}</li>
 *   <li>{@link #getCommits}</li>
 * </ul>
 *
 * <p>For example, semantic equality might be based on both instances pointing to the same 
 * underlying endpoint.</p>
 *
 * @param other The {@link CommitCoordinatorClientHandler} to compare with this instance
 * @return Boolean True if the instances are semantically equal, false otherwise
 */
```